src/components/TodoItem.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  changingIDs: number[];
+  onDelete?: (id: number) => Promise<void>;
+  onToggle?: (id: number, completed: boolean) => void;
+  onRename?: (id: number, title: string) => Promise<void>;
+  onEmptyTitleDelete?: () => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  changingIDs,
+  onDelete = () => Promise.resolve(),
+  onToggle = () => {},
+  onRename = () => Promise.resolve(),
+  onEmptyTitleDelete = () => {},
```

Comment:
Why do you need these defaults?

Reply:
for prevent errors

Reply:
Ok, a few more notes

1. You have already made these props optional - so no point pass the defaults. Just add checks for undefined (via `?.`) and it will solve your problem much cleaner
2. `onEmptyTitleDelete` looks an overkill - how exactly it is different from `onDelete`?